tools/Google.Cloud.Tools.DocUploader/Program.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using CommandLine;
+using Google.Apis.Auth.OAuth2;
+using Google.Cloud.Storage.V1;
+using Google.Cloud.Tools.DocUploader;
+using Google.Protobuf;
+using Google.Protobuf.WellKnownTypes;
+using SharpCompress.Archives;
+using SharpCompress.Archives.Tar;
+using SharpCompress.Common;
+using SharpCompress.Writers.Tar;
+
+Parser.Default.ParseArguments<CreateMetadataOptions, UploadFileOptions>(args)
+    .WithParsed<CreateMetadataOptions>(options => CreateMetadata(options))
+    .WithParsed<UploadFileOptions>(options => CompressAndUploadFile(options));
+
+// Creates the Metadata object based on the specified options and then store at options.Destination.
+void CreateMetadata(CreateMetadataOptions options)
+{
+    var metadata = new Metadata
+    {
+        UpdateTime = Timestamp.FromDateTimeOffset(DateTimeOffset.UtcNow),
+        Name = options.PackageName,
+        Version = options.version,
+        Language = options.Language,
+        DistributionName = options.DistributionName,
+        ProductPage = options.ProductPage,
+        GithubRepository = options.GitHubRepo,
+        IssueTracker = options.IssueTracker,
+        Stem = options.Stem,
+    };
+    metadata.Xrefs.Add(options.Xrefs);
+    metadata.XrefServices.Add(options.XrefServices);
+
+    File.WriteAllText(options.Destination, metadata.ToString());
+}
+
+// Uploads the zip file to the specified storage bucket.
+void CompressAndUploadFile(UploadFileOptions options)
+{
+    string tempFileName = "temp.tar.gz";
+    CompressDirectory(options.DocumentationPath, tempFileName);
+    UploadFile(tempFileName, GetFilenameFromMetadata(), options.StagingBucket, options.Credentials);
+
+    string GetFilenameFromMetadata()
+    {
+        string metadataFilePath = string.IsNullOrEmpty(options.MetadataFile) ? Path.Combine(options.DocumentationPath, "docs.metadata.json") : options.MetadataFile;
```

Comment:
It would be weird to specify an empty string - I think it's fine to use `options.MetadataFile ?? Path.Combine(options.DocumentationPath, "docs.metadata.json")`

Reply:
done